include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -3091,33 +3046,48 @@ namespace internal
     {
       const unsigned int face_direction = face_no / 2;
 
-      evaluate_in_face_apply<0, EvalNormal, EvalTangent>(
-        values_dofs,
-        fe_eval,
-        scratch_data,
-        evaluation_flag,
-        face_direction,
-        subface_index,
-        std::integral_constant<bool, integrate>());
+      // We first evaluate the anisotropic faces, i.e the faces where
+      // face_direction != component. Note that the call order here is not
+      // important, since the pointers are shifted accordingly within the
+      // functio. However, this is the order in which the components will be in
+      // the quadrature points. Furthermore, the isotropic faces have no "normal
+      // direction" but we still pass in normal_dir = 2 since this is used for
+      // the pointers.
+      // -----------------------------------------------------------------------------------
+      // |          |                   Anisotropic faces                 | Isotropic faces|
+      // | Face dir | comp, coords, normal_dir | comp, coords, normal_dir | comp, coords   |
+      // | --------------------------------------------------------------------------------|
```

Comment:
Thanks for adding a table! I am still not 100% happy with the code: it does not match the table. 2D does not contain any `2`s; you remove the `2` division. I would suggest to get it on master, as it is and we clean up it later on.

Reply:
I left out `normal_dir` for the isotropic faces in the table since the parameter is not used as `normal_dir` for those cases, but only for shifting the pointers. I'm not sure how you would like it to be instead.